CMakeLists.txt

```diff
@@ -158,6 +158,7 @@ endif()
 find_package(GCRYPT REQUIRED)
 add_definitions(-DHAVE_GCRYPT)
 list(APPEND EXTRA_LIBS ${GCRYPT_LDFLAGS})
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${GCRYPT_CFLAGS}")
```

Comment:
I think it's better (or more "clean") to add definitions instead of changing directly CFLAGS, can you try that: `add_definitions(${GCRYPT_CFLAGS})`.
I think the add_definitions should work to compile tests too (which are written in C++).
If that work fine, please rebase your pull request with that instead.